tool/gravity/cli/install.go

```diff
@@ -418,49 +449,80 @@ type autojoinConfig struct {
 	systemDevice  string
 	dockerDevice  string
 	mounts        map[string]string
+	fromService   bool
+	serviceURL    string
+	advertiseAddr string
+	token         string
 }
 
-func autojoin(env *localenv.LocalEnvironment, environ LocalEnvironmentFactory, d autojoinConfig) error {
-	if err := checkRunningAsRoot(); err != nil {
+func autojoin(env *localenv.LocalEnvironment, environ LocalEnvironmentFactory, d autojoinConfig) (err error) {
+	if err := d.checkAndSetDefaults(); err != nil {
 		return trace.Wrap(err)
 	}
 
+	if !d.fromService {
+		err := updateJoinConfigFromCloudMetadata(context.TODO(), &d)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	env.Printf("auto joining to cluster %q via %v\n", d.clusterName, d.serviceURL)
```

Comment:
Nit: I'd make this consistent with the rest of our progress messages, i.e. use the method that prints the timestamp (`env.PrintStep()`) and capitalize.